Try adding "Stewart", a light blogging theme with a sidebar

kjellr opened the pull request:
This is a light Blockbase spinoff for blogging that includes a left sidebar. Still needs a little light polish (and a bunch of testing?), but it seems to work generally well as a block-based theme given the small amount of effort I've put into it. It's named "Stewart", after @ianstewart (who asked us to make a blogging theme with a sidebar). 

## Screenshot

![new-theme](https://user-images.githubusercontent.com/1202812/128405395-3bec34cf-35a9-4308-8e2d-e1993e4bd9b0.gif)

kjellr committed:
Try "Stewart" theme

kjellr commented:
I kept the left sidebar content really simple for now, so that this could theoretically work as a universal theme pretty easily. It just contains a navigation block (which we already have working for Quadrat for example), and the social links block. (Do we have a way to make a social nav work in both classic themes and block themes?). 

That said, to make this a _truly_ customizable universal sidebar theme we'd probably need to figure out a way to let non-FSE folks customize the sidebar with widgets. 😬

ianstewart commented:
Let me know when you need a beta tester. I'll switch up my blog on WordPress.com. I _have_ to use this theme _now_. :)

scruffian commented:
I wonder if we can use a "widget area" block to achieve this.

kjellr committed:
Add screenshot, update color scheme.

kjellr committed:
Add custom font.

kjellr committed:
Round out the templates used.

kjellr committed:
Merge branch 'trunk' into try/add-stewart-theme

kjellr committed:
Rearrange stuff.

kjellr commented:
Spent some time today making this a bit more of a "theme" and less of a blank starting point. I changed the colors and font, added a screenshot, and tidied up the templates. 

Before|After
---|---
![stewart-old](https://user-images.githubusercontent.com/1202812/129253805-ca2501bd-2d0a-40c1-b77c-52752d1a33b4.png)|![stewart-new](https://user-images.githubusercontent.com/1202812/129253810-61a8d655-175e-4b97-a8e7-2c8994ad9f09.png)

kjellr committed:
Change google fonts loading.

kjellr committed:
Move font family declaration

kjellr committed:
Correct google fonts loading.

kjellr committed:
Revise the tagName for the sidebar.

kjellr committed:
Tidy up some spacing.

kjellr committed:
Ensure each template has a main HTML element in it.

kjellr committed:
Merge branch 'trunk' into try/add-stewart-theme

kjellr committed:
Merge branch 'trunk' into try/add-stewart-theme

kjellr committed:
Remove blockbase dependency.

kjellr committed:
Get stylesheet working.

kjellr committed:
Fix menu.

kjellr committed:
Add missing styles.

kjellr committed:
Remove custom variables

kjellr committed:
Add archive template, fix footer spacing.

kjellr committed:
Add spacing.

kjellr commented:
Just a heads up that I've gone in and pushed a few commits to start separating this out from Blockbase. It should work just fine as a standalone block theme now, though there are some spacing inconsistencies and likely a few unnecessary theme.json styles. 

cc @jeffikus @scruffian

kjellr commented:
Known issues: 

- [x] Needs a font loading implementation.
- [x] Needs `theme.json` to be pruned of blockbase-only keys.
- [x] Needs consistent [top spacing](https://cloudup.com/c8jM4UYKgFI) for all templates. 
- [x] Needs to show all pages by default in the sidebar nav area. 
- [x] Figure out why the background color doesn't appear on the frontend.

kjellr committed:
Rename folders.

kjellr committed:
Remove extra CSS, fix up nav block.

kjellr committed:
Add fonts implementation.

kjellr commented on stewart/functions.php:
```diff
@@ -0,0 +1,71 @@
+		$font_families = 'Work+Sans:ital,wght@0,400;0,700;1,400;1,700';
+
+		$query_args = array(
+			'family' => $font_families,
```
[Usually](http://themeshaper.com/2014/08/13/how-to-add-google-fonts-to-wordpress-themes/) this `$font_families` variable gets wrapped in `urlencode()`, but for some reason that wasn't working for me? Could use a second look.

pbking replied:
We don't appear to urlencode the value in blockbase.  This seems right to me.

kjellr committed:
Standardize spacing across templates.

kjellr committed:
Fix navigation hover styles.

kjellr committed:
Adjust spacing.

kjellr committed:
Fix tagline spacing.

kjellr committed:
Fix color variable for social links.

kjellr committed:
Fix up hovers, color presets.

kjellr committed:
Tidy up comment form and buttons

kjellr committed:
Fix comments

kjellr committed:
Readjust tagline spacing

kjellr committed:
Increase logo size, readjust tagline.

kjellr committed:
Tidy up templates, add button hover style.

kjellr committed:
Tidy up theme.json

kjellr committed:
Tidy up sidebar spacing

kjellr committed:
Cleanup readme

kjellr committed:
Fix padding around site tagline.

kjellr commented:
This should be ready for review! I've tidied up all the templates and cleaned out the `theme.json`. 

Homepage|Single Post
---|---
![Screen Shot 2021-12-08 at 15 05 08](https://user-images.githubusercontent.com/1202812/145281115-dffcda4b-10db-4aa4-85cd-1613b3eb9653.png)|![Screen Shot 2021-12-08 at 15 05 18](https://user-images.githubusercontent.com/1202812/145281113-97291f43-0af2-4dda-8a34-e9dacc7bd8b7.png)

pbking commented on stewart/theme.json:
```diff
@@ -0,0 +1,361 @@
+					"name": "Foreground and background"
+				}
+			],
+			"palette": [
```
Looks like the palette colors `primary` `secondary` and `selection` aren't used at all, perhaps they should be removed to avoid confusion.

<img src="https://user-images.githubusercontent.com/146530/145284305-4870a4af-0840-410c-b387-af2d1a281873.png" width=200>

kjellr replied:
Primary is actually used in a couple places (like hovers). In any case, I just went in and rearranged these so they the most often used colors are up front. I also replaced "selection" with "tertiary" so that these colors match up with the ones we're using in our Blockbase themes. That way if the user has existing patterns that use these color names, they'll translate over just fine.

scruffian commented:
At "tablet" sizes the columns go to 50/50:
<img width="749" alt="Screenshot 2021-12-09 at 11 52 05" src="https://user-images.githubusercontent.com/275961/145391704-ed8be921-5788-4d23-a4de-4b35aadcf766.png">

This makes the right column quite small. I'm not sure if this is acceptable, but if its not I think we need to fix it in Gutenberg. I've encountered the same limitation in Remote too, so I think we should have a way to turn it off regardless. I added an issue here: https://github.com/WordPress/gutenberg/issues/37252

kjellr committed:
Revise colors.

kjellr committed:
Make underlines match screenshot.

kjellr committed:
Fix 50/50 columns enforcement on small screens.

kjellr committed:
Update mobile spacer size.

kjellr commented:
> This makes the right column quite small. I'm not sure if this is acceptable, but if its not I think we need to fix it in Gutenberg. I've encountered the same limitation in Remote too, so I think we should have a way to turn it off regardless. I added an issue here: WordPress/gutenberg#37252

Yeah, this is not ideal. I just pushed a quick fix that uses a custom classname and makes those columns 100% wide on small screens. Should be fine until that issue is addressed.

kjellr committed:
Remove unnecessary flex-grow property.

scruffian commented:
> Yeah, this is not ideal. I just pushed a quick fix that uses a custom classname and makes those columns 100% wide on small screens. Should be fine until that issue is addressed.

I'm concerned about doing this in CSS as it means that this block behaves differently to other column blocks. The feedback we've had in the past about using CSS is "CSS should not be used for structural changes".

scruffian commented:
We should explore adding our Google font stack to this...

kjellr commented:
> I'm concerned about doing this in CSS as it means that this block behaves differently to other column blocks. The feedback we've had in the past about using CSS is "CSS should not be used for structural changes".

Yeah, I agree. But I think this is mostly harmless? I don't feel strongly, so I'm happy to strip it out and let the editor do its wrong thing for now. If the columns are 50/50 on tablets for now, I don't consider it a blocker. (EDIT: Reverted for now in 3fa0b28)

> We should explore adding our Google font stack to this...

Yeah, probably through a plugin and/or some dotcom functionality though, so it can be re-used. Doesn't make sense to add that into every single theme.

kjellr committed:
Revert the columns CSS for now.

pbking commented:
This theme is completely crashing for me right now.  Can't figure out why.  No errors just a completely empty html payload.

Just this theme.  Not sure what's up.

skorasaurus commented:
> Yeah, probably through a plugin and/or some dotcom functionality though, so it can be re-used. Doesn't make sense to add that into every single theme.

I ran into it; requested at https://github.com/WordPress/gutenberg/issues/34752 with some short-term workarounds for it.

kjellr commented:
> This theme is completely crashing for me right now. Can't figure out why. No errors just a completely empty html payload.
> 
> Just this theme. Not sure what's up.

That's odd — @pbking what WP + Gutenberg version are you running? It's working fine for me w/Gutenberg 12.1 & WP 5.8.2.

pbking commented:
I was using Gutenberg 12.0.1 and WP 5.8.2
Testing again with Gutenber /trunk and the theme is working again.  So something to do with that slightly older version of Gutenberg I suppose...

Regardless it's working for me now.

kjellr commented:
Do you folks think this is in a decent-enough place where we can merge it in and get it up on dotcom for testing?

scruffian commented on stewart/parts/header.html:
```diff
@@ -0,0 +1,6 @@
+<!-- wp:group {"align":"full","layout":{"type":"flex","justifyContent":"space-between"},"style":{"spacing":{"padding":{"top":"var(--wp--custom--margin--horizontal, 30px)","bottom":"40px"}}},"className":"site-header"} -->
```
I think we should remove this class

MaggieCabrera replied:
then we are missing the responsive styles: https://github.com/Automattic/themes/blob/trunk/blockbase/sass/base/_header.scss

scruffian replied:
This isn't a Blockbase theme

MaggieCabrera replied:
sorry I didn't check!

scruffian commented on stewart/templates/header-footer-only.html:
```diff
@@ -0,0 +1,10 @@
+</main>
+<!-- /wp:group -->
+
+<!-- wp:template-part {"slug":"footer","tagName":"footer","layout":{"inherit":true},"className":"site-footer-container"} /-->
```
I think we should remove this class name too.

MaggieCabrera replied:
then we'll be missing the "sticky footer" styles https://github.com/Automattic/themes/blob/75542ea9cbaade68bc827b353bd8e23c8c69e8f8/blockbase/sass/base/_layout.scss#L12

scruffian replied:
This isn't a Blockbase child

scruffian commented on stewart/theme.json:
```diff
@@ -0,0 +1,361 @@
+				}
+			},
+			"core/post-date": {
+				"color": {
```
Setting these colors means that when a user updates the colors in the site editor, these ones stay the same.

scruffian commented on stewart/style.css:
```diff
@@ -0,0 +1,125 @@
+.wp-block-site-title a,
+.wp-block-post-title a,
+.wp-block-post-date a {
+	color: var(--wp--preset--color--foreground);
```
Do we need these lines?

scruffian commented:
The main problem I see is that when I customize colors they don't all get changed:
<img width="1288" alt="Screenshot 2021-12-14 at 11 10 01" src="https://user-images.githubusercontent.com/275961/145987642-ee75e86f-e1bd-4b19-85b2-c9cced2fe533.png">

kjellr committed:
Remove unnecessary classnames.

kjellr committed:
Remove unnecessary styles, change hover to opacity instead of a color.

kjellr commented:
@scruffian I've adjusted what I can, but some of this I think is just an issue with Global Styles. You used to be able to edit the "Primary" color, but that's not possible anymore (unless I'm missing something?).

scruffian commented:
Yeah that's what I was noticing too. Seems like we might need to open a Gutenberg issue for that as it's going to be a bit of a blocker...

kjellr commented:
@scruffian It's still there actually, I think we both actually just overlooked this control: 
![colors](https://user-images.githubusercontent.com/1202812/146061793-99b536d9-3b77-4f0f-bdbb-6157235baf40.gif)

kjellr committed:
Reinstate the hover colors.

scruffian reviewed (approved):
In that case I say LGTM

kjellr commented:
Awesome. I'll leave this up to you folks to merge and deploy. 👍

kjellr committed:
Update description.

kjellr committed:
Update description.
